tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/EmailRecoveryActionGetMethod.scala

```diff
@@ -0,0 +1,79 @@
+import play.api.libs.json.{JsError, JsObject}
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+class EmailRecoveryActionGetMethod @Inject()()(val taskManager: TaskManager,
+                                               val metricFactory: MetricFactory,
+                                               val sessionSupplier: SessionSupplier) extends MethodWithoutAccountId[EmailRecoveryActionGetRequest] {
+
+  override val methodName: Invocation.MethodName = MethodName("EmailRecoveryAction/get")
+
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_CORE, LINAGORA_MESSAGE_VAULT)
+
+  override def getRequest(invocation: Invocation): Either[Exception, EmailRecoveryActionGetRequest] =
+    EmailRecoveryActionSerializer.deserializeGetRequest(invocation.arguments.value).asEither
+      .left.map(errors => new IllegalArgumentException(ResponseSerializer.serialize(JsError(errors)).toString))
+
+  override def doProcess(invocation: InvocationWithContext, mailboxSession: MailboxSession, request: EmailRecoveryActionGetRequest): Publisher[InvocationWithContext] =
+    request.validateProperties
+      .fold(e => SMono.error(e),
+        properties => retrieveEmailRecoveryActions(mailboxSession.getUser, request.ids)
+          .collectSeq()
+          .map(emailRecoveryActions => EmailRecoveryActionGetResponse.from(emailRecoveryActions, request.ids))
+          .map(response => Invocation(
+            methodName = methodName,
+            arguments = Arguments(EmailRecoveryActionSerializer.serializeGetResponse(response, properties).as[JsObject]),
+            methodCallId = invocation.invocation.methodCallId))
+          .map(invocationResult => InvocationWithContext(invocationResult, invocation.processingContext)))
+
+  private def retrieveEmailRecoveryActions(username: Username, ids: EmailRecoveryActionIds): SFlux[EmailRecoveryAction] =
+    SFlux.fromIterable(filteringValidTaskIds(ids))
+      .flatMap(taskId => SMono.fromCallable(() => taskManager.getExecutionDetails(taskId))
+        .subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER))
+      .filter(taskDetail => isDeletedMessagesVaultRestoreTask(taskDetail) && belongsToUser(username, taskDetail))
+      .map(toEmailRecoveryAction)
+
+  private def filteringValidTaskIds(ids: EmailRecoveryActionIds): List[TaskId] =
+    ids.list.map(unparsedId => TaskIdUtil.liftOrThrow(unparsedId).toOption)
+      .filter(_.isDefined)
+      .map(_.get)
+
+  private def isDeletedMessagesVaultRestoreTask(taskDetail: TaskExecutionDetails): Boolean =
+    taskDetail.getType.equals(DeletedMessagesVaultRestoreTask.TYPE)
+
+  private def belongsToUser(username: Username, taskDetail: TaskExecutionDetails): Boolean =
+    taskDetail.getAdditionalInformation.get()
+      .asInstanceOf[DeletedMessagesVaultRestoreTask.AdditionalInformation]
+      .getUsername
+      .equals(username.asString())
+
+  private def toEmailRecoveryAction: TaskExecutionDetails => EmailRecoveryAction =
```

Comment:
```suggestion
  private def toEmailRecoveryAction(taskDetails: TaskExecutionDetails): EmailRecoveryAction = { ... } 
```

Less boiler plate IMO